lib/deprecation.nix

```diff
@@ -46,6 +46,21 @@ rec {
         '';
     };
 
+  processRenameOptions =
+    options: oldPath: newPath:
```

Comment:
I'd rather re-arrange the arguments so that `options` is last.

In fact, if we drop the unnecessary parentheses, we could drop the explicit `options` argument entirely and just return `map (x: /* omitted */)`, which will implicitly add the additional arg by returning a curried `map` function.

```nix
oldPath: newPath:
map (
  option:
  let
    # omitted
  in
    # omitted
);
```

Reply:
On the subject of arguments, it might be better to use `oldPrefix` and `newPrefix`, allowing us to do this within the mapping function:

```nix
let
  optionPath = toList option;
  oldPath = oldPrefix ++ optionPath;
  newPath = newPrefix ++ toSnakeCase optionPath;
in
```